### ISSUE TITLE
No support for etcd TLS server-only authentication

### ISSUE THREAD

**Issue by TrimBiggs** [kelseyhightower/confd#364]
There is support for creating an etcd client object that can access a TLS etcd server with client certificate authentication.  However, there does not seem to be support for accessing a TLS etcd server with JUST server CA authentication (eg. [Etcd Security Model Example 1](https://github.com/coreos/etcd/blob/master/Documentation/security.md#example-1-client-to-server-transport-security-with-https)).

I found this by running an etcd server, passing in `--cert-file "/etc/certs/server.crt"  --key-file "/etc/certs/server.key"`.  I can access the server just fine using `curl` or `etcdctl` by passing in the CA certificate for the CA that signed the server's key pair.  With confd, I see an error saying that it cannot access etcd.

I can see in [`client.go`](https://github.com/kelseyhightower/confd/blob/bec0b60ec4331f3a3ee1db104f46983a6f035798/src/github.com/kelseyhightower/confd/backends/etcd/client.go#L18) that a NewEtcdClient can be creating with `caCert`, but this also requires a `key` and `cert` for TLS.  There should be a way to just specify the CA certificate without key/cert, as referenced in the main etcd documentation.

I've also raised an issue with coreos/goetcd, https://github.com/coreos/go-etcd/issues/248, since I imagine this is partly due to goetcd not supporting this yet.

**Comment by jonboulle** [comment]
go-etcd is deprecated, so this is unlikely to be fixed here; the real solution is to migrate confd to the official [`github.com/coreos/etcd/client`](https://github.com/coreos/etcd/tree/master/client) client (which is use by etcdctl)

**Comment by bacongobbler** [comment]
related: https://github.com/kelseyhightower/confd/issues/345

**Comment by jonboulle** [comment]
Oops, I guess i put my comment on the wrong repository.

**Comment by HeavyHorst** [comment]
Should be fixed in #392 

```
confd -log-level=debug -watch -backend=etcd -node="https://localhost:4001" -client-ca-keys="ca.pem"
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: INFO Backend set to etcd
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: INFO Starting confd
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: INFO Backend nodes set to https://localhost:4001
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Loading template resources from confdir /etc/confd
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Found template: /etc/confd/conf.d/myconfig.toml
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Loading template resource from /etc/confd/conf.d/myconfig.toml
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Retrieving keys from store
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Key prefix set to /
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Using source template /etc/confd/templates/myconfig.conf.tmpl
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Compiling source template /etc/confd/templates/myconfig.conf.tmpl
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Comparing candidate config to /tmp/myconfig.conf
2016-02-19T19:38:51+01:00 rkaufmann confd[1177]: DEBUG Target config /tmp/myconfig.conf in sync
```

**Comment by bacongobbler** [comment]
fixed with https://github.com/kelseyhightower/confd/pull/387.
